spec.bs

```diff
@@ -1159,12 +1373,208 @@ fields:
+        -7 ("ES256").
+    * {{PublicKeyCredentialParameters/type}} =
+        {{PublicKeyCredentialType}} with {{PublicKeyCredentialParameters/alg}} =
+        -257 ("RS256").
+
+1. Remove any entries from |allowed_algorithms| that are not of type
+    {{PublicKeyCredential}}.
+
+1. Remove any entries from |allowed_algorithms| that the user agent does
+    not support.
+
+    Note: User agents may support different sets of a cryptographic algorithms
+    on different device platforms.
+
+1. If |allowed_algorithms|'s [=list/size=]
+
+    <dl class="switch">
+        : is zero
+        :: Stop, returning null.
+
+            Note: The user agent does not add any browser bound outputs in this
+            case.
+
+        : is non-zero
+        :: Proceed to create the key pair:
+
+            1. Let |chosen_algorithm| be the first entry of
+                |allowed_algorithms|.
+
+            1. Generate the public private key pair using the established
+                procedure corresponding to |chosen_algorithm|.
+
+            1. If the step above failed for any reason, return null.
+
+            1. Let |bbk| be the the key pair from the key generation step above.
+
+            1. Generate an identifier, |bbk_id|.
+
+                Note: Many cryptographic APIs may return an identifier, handle,
+                    or wrapped key instead of the private key and along with
+                    the public key.
+                    For example when the private key is stored
+                    in a secure element, the private key is not directly
+                    available to the user agent, and the
+                    identifier/handle/wrapped_key must be passed into another
+                    function of the cryptographi API in order to be used.
+
+            1. Return the tuple (|bbk|, |bbk_id|).
+
+    </dl>
+
+Issue: The spec could specify to prefer algorithms in hardware storage (in
+    given order) then algorithms in software (in the same given order). For the
+    time being this topic may be moot since Chrome plans to support only 1
+    algorithm per platform in hardware. See [Secure Payment Confirmation issue
+    #288](https://github.com/w3c/secure-payment-confirmation/issues/288) that
+    discusses storage type of BBKs including additional topics of which storage
+    types should be allowed and exposing the storage type to the relying party.
+
+## Binding a key pair ## {#sctn-binding-a-keypair}
+
+To <dfn>bind a key pair</dfn> given the |bbk_id|, the key pair identifier, and
+|credential_id|, the SPC credential_id (i.e. passkey id), store the
+identifiers in the [=browser_bound_map=] as follows:
+
+1. Insert the browser bound key id into the map:
+    [=browser_bound_map=][|credential_id|] = |bbk_id|.
+
+    Note: When user agents are storing this map (such as serializing to disk)
+        and the procedure fails, the user agent may return an error
```

Comment:
This is normative, so it cannot be buried inside a note. You need an explicit step saying to handle any implemetation-defined errors by returning "failure" or returning null, or something else that the callers can explicitly key their behavior off of. But the point is, you have to account for that failure case explicitly in this algorithm.